### PR TITLE
Code cleanup

### PR DESCRIPTION
**Pull request by dkeeney**
- removed ptr_types.hpp, 
- removed NodeSet.hpp, 
- combined Region.cpp, RegionIo.cpp, RegionParameters.cpp

### CODE REVIEW

**Comment by dkeeney** [comment]
We have a failure in debug mode in circleci.  I need some help here.
This PR has no functional or logic changes so whatever is causing this was there before.
It seems that every time I run this test the results are wildly different.
```
[----------] 1 test from HelloRegionTest
[ RUN      ] HelloRegionTest.demo
Setting region dimensions[1]
Loading data from /Users/distiller/numenta/nupic.core/src/test/unit/engine/Data.csv
In VectorFileSensor /Users/distiller/numenta/nupic.core/src/test/unit/engine/Data.csv
INFO:   appendFile - need to reset scale and offset vectors.
Read 4 vectors
Initializing network
Compute
  0    -2.31584e+77
In VectorFileSensor /Users/distiller/numenta/nupic.core/src/test/unit/engine/Data.csv
INFO:   appendFile - need to reset scale and offset vectors.
Read 4 vectors
/Users/distiller/numenta/nupic.core/src/test/unit/engine/HelloRegionTest.cpp:114: Failure
The difference between buffer[i] and buffer2[i] is 5.5429650347661715e+70, which exceeds 0.001, where
buffer[i] evaluates to -2.3158417847463239e+77,
buffer2[i] evaluates to -2.3158423390428274e+77, and
0.001 evaluates to 0.001.
 buffer -2.31584e+77 buffer2: -2.31584e+77
[  FAILED  ] HelloRegionTest.demo (1 ms)
```

**Comment by dkeeney** [comment]
This time HelloRegionTest.demo happens to work.

**Comment by dkeeney** [comment]
This time HelloRegionTest.demo resulted in:
Read 4 vectors
 buffer -0 buffer2: -5.32571e-315
This does not look right to me but I do not know where to start looking.

**Comment by ctrl-z-9000-times** [comment]
Yea the numbers look wrong, especially since IIRC the datafile contains integers 0,1,2,3.

File: bindings/py/cpp_src/bindings/engine/py_Engine.cpp [hunk 1]
```diff
@@ -52,6 +52,9 @@ using namespace nupic;
 
 namespace nupic_ext
 {
+
+	typedef std::shared_ptr<Region> Region_Ptr_t;
```
Comment (ctrl-z-9000-times) [inline bindings/py/cpp_src/bindings/engine/py_Engine.cpp]: nitpick: replace tabstops with 4 spaces.

File: bindings/py/cpp_src/plugin/RegisteredRegionImplPy.hpp [hunk 2]
```diff
@@ -146,6 +146,7 @@ namespace nupic
 			PyBindRegion::createSpec(module_.c_str(), *sp, classname_.c_str());
 		  }
           catch (nupic::Exception & e) {
+		    UNUSED(e);
```
Comment (ctrl-z-9000-times) [inline bindings/py/cpp_src/plugin/RegisteredRegionImplPy.hpp]: nitpick: replace tabstops with 4 spaces.

File: bindings/py/cpp_src/bindings/engine/py_Engine.cpp (hunk 1 above)
Comment (dkeeney) [reply bindings/py/cpp_src/bindings/engine/py_Engine.cpp]: > nitpick: replace tabstops with 4 spaces.

yeah, we have not been very consistent on that.  Many of the modules use 2 space indent...or even 1 space.

**Comment by dkeeney** [comment]
Thanks.

**Pull request merged**